packages/pac-proxy-agent/src/index.ts

```diff
@@ -279,6 +302,9 @@ export class PacProxyAgent<Uri extends string> extends Agent {
 						);
 					}
 					req.emit('proxy', { proxy, socket: s });
+					if (agent instanceof HttpProxyAgent) {
+						this.httpProxyAgent = agent;
+					}
```

Comment:
This is what I had to do to fix `keepAlive` for HTTP proxies in "pac-proxy-agent"`.

Essentiallly:

* When the socket is successfully created, if it was made with `HttpProxyAgent`, then that child agent is cached as `this.httpProxyAgent`.
* On subsequent requests, if `this.httpProxyAgent` is defined, that is used to rewrite the request path and add headers.

Reply:
I removed this change in bac30180bd8e629dba7393968d05786ceede0768 in order to descope this PR